src/attachments/receiving/receiving.ts

```diff
@@ -34,8 +102,43 @@ export class Receiving {
       ? `/emails/receiving/${emailId}/attachments?${queryString}`
       : `/emails/receiving/${emailId}/attachments`;
 
-    const data = await this.resend.get<ListAttachmentsResponseSuccess>(url);
+    const apiResponse = await this.resend.get<ListAttachmentsApiResponse>(url);
+
+    if ('error' in apiResponse && apiResponse.error) {
+      return apiResponse;
+    }
+
+    const attachmentsWithContent = [];
+    for (const attachment of apiResponse.data.data) {
+      const {
+        expires_at: _expires_at,
+        download_url,
+        ...otherFields
+      } = attachment;
+      const downloadResult = await this.downloadAttachment(download_url);
```

Comment:
Not sure if performance is a concern for this, but you might want to put this under a `Promise.all` or `Promise.allSettled` so that all the attachments are downloaded in parallel. 

I missed that during my first review, sorry 😥